bigquery/google/cloud/bigquery/job.py

```diff
@@ -676,6 +698,11 @@ def from_api_repr(cls, resource):
         config.skip_leading_rows = _int_or_none(slr)
         if config.skip_leading_rows is None:
             del config.skip_leading_rows
+        if ('destinationEncryptionConfiguration' in resource
```

Comment:
Is it possible that there will be other `EncryptionConfiguration`s? If so, wouldn't it be unwise for us to reserve `destinationEncryptionConfiguration`'s `kmsKeyName` as the top-level `kms_key_name` property?

Reply:
I asked that in the issue, also. https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4842

You're right. I'd wager there's going to be other `EncryptionConfigurations`. We should probably make a separate class to hold the various ones?

Reply:
Refactored. Added EncryptionConfiguration class and respective properties.